src/main/cpp/t04_both2.cpp

```diff
@@ -23,9 +23,34 @@
 
 #include "t04_both2.h"
 #include <iostream>
+#include <set>
 
 using namespace std;
 
 int t04_both2() {
-
-}
+    set <int>z;
+    set <int>z2;
+    set <int>z1;
+    int q;
+    int w;
+    cin >> q;
+    for(int i = 0; i < q; i++)
+    {
+        int a;
+        cin >> a;
+        z.insert(a);
+    }
+    cin >> w;
+    for(int i = 0; i < w; i++)
+    {
+        int a;
+        cin >> a;
+        z1.insert(a);
+        if(z.find(a) != z.end()) {
+            z2.insert(a);
+        }
+    }
+    set <int>::iterator j;
+    for(j = z2.begin(); j != z2.end(); j++)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer prefix ++/-- operators for non-primitive types.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4688993)